aries_cloudagent/revocation/routes.py

```diff
@@ -0,0 +1,242 @@
+"""Revocation registry admin routes."""
+
+from asyncio import shield
+
+from aiohttp import web
+from aiohttp_apispec import docs, request_schema, response_schema
+
+from marshmallow import fields, Schema
+
+from ..messaging.credential_definitions.util import CRED_DEF_SENT_RECORD_TYPE
+from ..messaging.valid import INDY_CRED_DEF_ID
+from ..storage.base import BaseStorage, StorageNotFoundError
+
+from .error import RevocationNotSupportedError
+from .indy import IndyRevocation
+from .models.issuer_revocation_record import IssuerRevocationRecordSchema
+from .models.revocation_registry import RevocationRegistry
+
+class RevRegCreateRequestSchema(Schema):
+    """Request schema for revocation registry creation request."""
+
+    credential_definition_id = fields.Str(
+        description="Credential definition identifier", **INDY_CRED_DEF_ID
+    )
+
+    max_cred_num = fields.Int(
+        description="Maximum credential numbers", required=False
+    )
+
+
+class RevRegCreateResultSchema(Schema):
+    """Result schema for revocation registry creation request."""
+
+    result = IssuerRevocationRecordSchema()
+
+
+class RevRegUpdateTailFileUriSchema(Schema):
+    """Request schema for updating tail file URI."""
+
+    tails_public_uri = fields.Url(
+        description="Public URI to the tail file", required=True
+    )
+
+
+@docs(tags=["revocation"], summary="Creates a new revocation registry")
+@request_schema(RevRegCreateRequestSchema())
+@response_schema(RevRegCreateResultSchema(), 200)
+async def revocation_create_registry(request: web.BaseRequest):
+    """
+    Request handler for creating a new revocation registry.
+
+    Args:
+        request: aiohttp request object
+
+    Returns:
+        The revocation registry identifier
+
+    """
+    context = request.app["request_context"]
+
+    body = await request.json()
+
+    credential_definition_id = body.get("credential_definition_id")
+    max_cred_num = body.get("max_cred_num")
+
+    # check we published this cred def
+    storage = await context.inject(BaseStorage)
+    found = await storage.search_records(
+        type_filter=CRED_DEF_SENT_RECORD_TYPE,
+        tag_query={"cred_def_id": credential_definition_id},
+    ).fetch_all()
+    if not found:
+        raise web.HTTPNotFound()
+
+    try:
+        issuer_did = credential_definition_id.split(":")[0]
+        revoc = IndyRevocation(context)
+        registry_record = await revoc.init_issuer_registry(
+            credential_definition_id, issuer_did, max_cred_num=max_cred_num
+        )
+    except RevocationNotSupportedError as e:
+        raise web.HTTPBadRequest(reason=e.message) from e
+    await shield(
+        registry_record.generate_registry(context, RevocationRegistry.get_temp_dir())
```

Comment:
This operation can take minutes - consider a background process that generates the revocation registry ahead of time, with tag increment.

Alternatively, the non-secrets API could hold a work queue.